### PR TITLE
Config file handling

### PR DESCRIPTION
**Pull request by tnking97**
Improve config file handling as described in #634

### CODE REVIEW

**Comment by bourque** [comment]
@tnking97 FYI, as I was doing some work with #726, I realized that we have `utils._validate_config` and `test_utils.test_validate_config` functions that make sure the `config.json` file is structured in an expected way.  This PR would be a good opportunity to make sure those functions are up-to-date with the current config file.  I think there may have been some keys added since those functions were originally written.

**Comment by bhilbert4** [comment]
Here's a possibly-related issue with the config file relating to #711. When I looked into the log of the [latest documentation build on readthedocs](https://readthedocs.org/api/v2/build/13747438.txt), I noticed there are a bunch of FileNotFoundError, where the config file isn't being found by readthedocs when building the environment. This is preventing some JWQL modules from being imported, and it looks like these are the modules for which the API documentation pages are not being built.

File: example_config.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (bourque) [inline example_config.json:1]: Great idea to name this something different than `config.json` so that we avoid committing personal config files!

**Review by bourque — APPROVED** [review]
Thank you @tnking97, this looks great!  And thank you for updating the wiki page too!

**Pull request merged**